app/src/main/java/com/raghav/spacedawn/ui/AppViewModel.kt

```diff
@@ -1,26 +1,28 @@
 package com.raghav.spacedawn.ui
 
-import android.app.Application
 import android.content.Context
 import android.net.ConnectivityManager
 import android.net.NetworkCapabilities
-import androidx.lifecycle.AndroidViewModel
 import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
 import com.raghav.spacedawn.db.ReminderModelClass
 import com.raghav.spacedawn.models.launchlibrary.LaunchLibraryResponse
 import com.raghav.spacedawn.models.spaceflightapi.ArticlesResponse
-import com.raghav.spacedawn.repository.AppRepository
+import com.raghav.spacedawn.repository.IAppRepository
 import com.raghav.spacedawn.utils.AppApplication
 import com.raghav.spacedawn.utils.Resource
+import dagger.hilt.android.lifecycle.HiltViewModel
 import kotlinx.coroutines.launch
 import retrofit2.Response
 import java.io.IOException
+import javax.inject.Inject
 
-class AppViewModel(
-    app: Application,
-    val repository: AppRepository
-) : AndroidViewModel(app) {
+@HiltViewModel
+class AppViewModel @Inject constructor(
```

Comment:
lets plan on fragments having there own view models and only shared states residing in this one